<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InboundEditor</class>
 <widget class="QDialog" name="InboundEditor">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>815</width>
    <height>489</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Inbound Editor</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="7,3">
   <item row="2" column="0">
    <layout class="QFormLayout" name="formLayout_4">
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Tag</string>
       </property>
       <property name="buddy">
        <cstring>inboundTagTxt</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="inboundTagTxt">
       <property name="placeholderText">
        <string>Tag of this inbound entry</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Listening</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="2,0,1">
       <item>
        <widget class="QLineEdit" name="inboundHostTxt">
         <property name="placeholderText">
          <string>Hostname or IP Address</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="inboundPortTxt">
         <property name="placeholderText">
          <string>Port: 1080|80-85</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Protocol</string>
       </property>
       <property name="buddy">
        <cstring>inboundProtocolCombo</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="inboundProtocolCombo">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string notr="true">http</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">socks</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">dokodemo-door</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">mtproto</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Allocation Settings</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Strategy</string>
        </property>
        <property name="buddy">
         <cstring>strategyCombo</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="strategyCombo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>always</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>random</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Refresh</string>
        </property>
        <property name="buddy">
         <cstring>refreshNumberBox</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="refreshNumberBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Concurrency</string>
        </property>
        <property name="buddy">
         <cstring>concurrencyNumberBox</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="concurrencyNumberBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Sniffing Settings</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>32</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>80</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Destination Override</string>
        </property>
        <property name="textFormat">
         <enum>Qt::RichText</enum>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="buddy">
         <cstring>destOverrideList</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QListWidget" name="destOverrideList">
        <item>
         <property name="text">
          <string notr="true">HTTP</string>
         </property>
         <property name="checkState">
          <enum>Unchecked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">TLS</string>
         </property>
         <property name="checkState">
          <enum>Unchecked</enum>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="enableSniffingCB">
        <property name="text">
         <string>Enabled</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_29">
        <property name="text">
         <string>Enabled</string>
        </property>
        <property name="buddy">
         <cstring>enableSniffingCB</cstring>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="HTTPPage">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>HTTP Inbound Settings</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,6">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Timeout</string>
              </property>
              <property name="buddy">
               <cstring>httpTimeoutSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="httpTimeoutSpinBox"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Allow Transparent</string>
              </property>
              <property name="buddy">
               <cstring>httpTransparentCB</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="httpTransparentCB">
              <property name="text">
               <string>Enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>User Level</string>
              </property>
              <property name="buddy">
               <cstring>httpUserLevelSB</cstring>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="httpUserLevelSB"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,1,1,0">
            <item row="1" column="3">
             <widget class="QToolButton" name="httpRemoveUserBtn">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" rowspan="2" colspan="3">
             <widget class="QListWidget" name="httpAccountListBox"/>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="httpAddPasswordTxt">
              <property name="placeholderText">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QToolButton" name="httpAddUserBtn">
              <property name="text">
               <string>+</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Add</string>
              </property>
              <property name="buddy">
               <cstring>httpAddUserTxt</cstring>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="httpAddUserTxt">
              <property name="placeholderText">
               <string>Username</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="5">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Accounts</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="SOCKSPage">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>SOCKS Inbound Settings</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="4,6">
          <item>
           <layout class="QFormLayout" name="formLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Auth</string>
              </property>
              <property name="buddy">
               <cstring>socksAuthCombo</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="socksAuthCombo">
              <item>
               <property name="text">
                <string notr="true">noauth</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">password</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Enable UDP</string>
              </property>
              <property name="buddy">
               <cstring>socksUDPCB</cstring>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>Local UDP IP</string>
              </property>
              <property name="buddy">
               <cstring>socksUDPIPAddrTxt</cstring>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_21">
              <property name="text">
               <string>User Level</string>
              </property>
              <property name="buddy">
               <cstring>socksUserLevelSB</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="socksUDPCB">
              <property name="text">
               <string>Enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="socksUDPIPAddrTxt">
              <property name="placeholderText">
               <string>127.0.0.1</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="socksUserLevelSB"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_4" rowstretch="0,1,1,0">
            <item row="1" column="3">
             <widget class="QToolButton" name="socksRemoveUserBtn">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" rowspan="2" colspan="3">
             <widget class="QListWidget" name="socksAccountListBox"/>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="socksAddPasswordTxt">
              <property name="placeholderText">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QToolButton" name="socksAddUserBtn">
              <property name="text">
               <string>+</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>Add</string>
              </property>
              <property name="buddy">
               <cstring>socksAddUserTxt</cstring>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="socksAddUserTxt">
              <property name="placeholderText">
               <string>Username</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="5">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Accounts</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="DokodemoPage">
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Dokodemo-Door Inbound Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_6">
          <item row="1" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>IP Address</string>
            </property>
            <property name="buddy">
             <cstring>dokoIPAddrTxt</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="dokoIPAddrTxt">
            <property name="placeholderText">
             <string>Not necessary when setting &quot;Follow Redirect&quot;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Port</string>
            </property>
            <property name="buddy">
             <cstring>dokoPortSB</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="dokoPortSB">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>10086</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Network</string>
            </property>
            <property name="buddy">
             <cstring>dokoTCPCB</cstring>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QCheckBox" name="dokoTCPCB">
              <property name="text">
               <string>TCP</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="dokoUDPCB">
              <property name="text">
               <string>UDP</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>Timeout</string>
            </property>
            <property name="buddy">
             <cstring>dokoTimeoutSB</cstring>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Follow Redirect</string>
            </property>
            <property name="buddy">
             <cstring>dokoFollowRedirectCB</cstring>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>User Level</string>
            </property>
            <property name="buddy">
             <cstring>dokoUserLevelSB</cstring>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="dokoTimeoutSB">
            <property name="minimum">
             <number>-1</number>
            </property>
            <property name="maximum">
             <number>8192</number>
            </property>
            <property name="value">
             <number>300</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="dokoFollowRedirectCB">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QSpinBox" name="dokoUserLevelSB"/>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_30">
            <property name="text">
             <string>If you want to use tProxy, please go to Preference Window to enable this feature.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="MTProtoPage">
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>MTProto Inbound Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_7">
          <item row="0" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>EMail Address</string>
            </property>
            <property name="buddy">
             <cstring>mtEMailTxt</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mtEMailTxt">
            <property name="placeholderText">
             <string notr="true">your.name@domain.com</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>Secret</string>
            </property>
            <property name="buddy">
             <cstring>mtSecretTxt</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="mtSecretTxt">
            <property name="placeholderText">
             <string>SECRET</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>User Level</string>
            </property>
            <property name="buddy">
             <cstring>mtUserLevelSB</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="mtUserLevelSB"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>inboundTagTxt</tabstop>
  <tabstop>inboundHostTxt</tabstop>
  <tabstop>inboundPortTxt</tabstop>
  <tabstop>inboundProtocolCombo</tabstop>
  <tabstop>httpTimeoutSpinBox</tabstop>
  <tabstop>httpTransparentCB</tabstop>
  <tabstop>httpUserLevelSB</tabstop>
  <tabstop>httpAccountListBox</tabstop>
  <tabstop>httpRemoveUserBtn</tabstop>
  <tabstop>httpAddUserTxt</tabstop>
  <tabstop>httpAddPasswordTxt</tabstop>
  <tabstop>httpAddUserBtn</tabstop>
  <tabstop>socksAuthCombo</tabstop>
  <tabstop>socksUDPCB</tabstop>
  <tabstop>socksUDPIPAddrTxt</tabstop>
  <tabstop>socksUserLevelSB</tabstop>
  <tabstop>socksAccountListBox</tabstop>
  <tabstop>socksRemoveUserBtn</tabstop>
  <tabstop>socksAddUserTxt</tabstop>
  <tabstop>socksAddPasswordTxt</tabstop>
  <tabstop>socksAddUserBtn</tabstop>
  <tabstop>dokoIPAddrTxt</tabstop>
  <tabstop>dokoPortSB</tabstop>
  <tabstop>dokoTCPCB</tabstop>
  <tabstop>dokoUDPCB</tabstop>
  <tabstop>dokoTimeoutSB</tabstop>
  <tabstop>dokoFollowRedirectCB</tabstop>
  <tabstop>dokoUserLevelSB</tabstop>
  <tabstop>mtEMailTxt</tabstop>
  <tabstop>mtSecretTxt</tabstop>
  <tabstop>mtUserLevelSB</tabstop>
  <tabstop>strategyCombo</tabstop>
  <tabstop>refreshNumberBox</tabstop>
  <tabstop>concurrencyNumberBox</tabstop>
  <tabstop>enableSniffingCB</tabstop>
  <tabstop>destOverrideList</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>InboundEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>InboundEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
